File: handlerclaw/lib/features/finances/data/dto/account_dto.dart
class AccountDto {
  final String id;
  final String name;
  final String category;
  final double currentAmount;
  final String accountTypeId;

  AccountDto({
    required this.id,
    required this.name,
    required this.category,
    required this.currentAmount,
    required this.accountTypeId,
  });

  factory AccountDto.fromJson(Map<String, dynamic> json) {
    return AccountDto(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      category: json['category'] ?? 'LIQUID',
      currentAmount: (json['current_amount'] ?? 0).toDouble(),
      accountTypeId: json['account_type_id'] ?? '',
    );
  }
}

File: api/src/modules/(jwt-auth)/finances/accounts/application/dtos/account.dto.ts
import type { PaginationMetaDto } from "../../../../../../lib/types/pagination-meta-dto.js";
import type { IAccount } from "../../domain/entities/account.entity.js";
export type AccountDto = Omit<IAccount, "balances"> & {
  current_amount: number;
  category: string;
};

/**
 * Input Data Contracts
 */
export type CreateAccountData = {
  name: string;
  account_type_id: string;
};

export type CreateAccountRequest = {
  name: string;
  account_type_id: string;
};

export type UpdateAccountRequest = {
  name?: string | undefined;
  account_type_id?: string | undefined;
};

export type GetAccountsQuery = {
  page: number;
  per_page: number;
  search?: string | undefined;
};

/**
 * Response Contracts
 */
export type CreateAccountResponse = {
  account: AccountDto;
};

export type GetAccountsResponse = {
  accounts: AccountDto[];
  meta: PaginationMetaDto;
};

export type GetAccountDetailResponse = {
  account: AccountDto;
};

export type UpdateAccountResponse = {
  account: AccountDto;
};
